File: Array/8 Faulty Keyboard/ques8.cpp
class Solution {
public:
    string finalString(string s) {
        string res;

        for (int c : s){
            if (c=='i'){
                reverse(res.begin(), res.end());
            }

            else {
                res+=c;
            }
        }
        return res;
    }
};

File: Array/8 Faulty Keyboard/ques8.js
/**
 * @param {string} s
 * @return {string}
 */
var finalString = function(s) {
    let res="";
    
    for (let i=0; i< s.length; i++){
        if (s[i]==='i'){
            res=res.split("").reverse().join("")
        }
        else {
            res+=s[i]
        }
    }
    
    return res
    };